src/plugins/charts/DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,180 @@
+# Usage
```

Comment:
Yeah, i'm not convinced this is the right place to have this information. Seems like the forked docsite might be a better place, but not sure how maintained that is currently. 

Didnt others create similar docs on how other plugins work, e.g. How visualizations work using a flowchart. where did we share that?

One other alternative to pave the way could be to just have it here but call it something more appropriate like `Current_useage.md`. Developer guide is what threw me off the most. That usually means something else.

Reply:
OK, in this new revision, I've moved to the `docs` directory and renamed.

> Didnt others create similar docs on how other plugins work, e.g. How visualizations work using a flowchart. where did we share that?

Not that I know of? I think some information like that is simply in GitHub issues, such as https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2187

Reply:
I like that! That seems like a better place for high level information like this for now while we decide on the best place to consolidate this.